iroh/src/rpc_protocol.rs

```diff
@@ -177,6 +208,41 @@ impl ServerStreamingMsg<RpcService> for BlobValidateRequest {
     type Response = ValidateProgress;
 }
 
+/// Get the status of a blob
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BlobStatusRequest {
+    /// The hash of the blob
+    pub hash: Hash,
+}
+
+/// The response to a status request
+#[derive(Debug, Serialize, Deserialize, derive_more::From, derive_more::Into)]
+pub struct BlobStatusResponse(pub BlobStatus);
+
+/// Status information about a blob.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum BlobStatus {
```

Comment:
lets put this into the `blobs.rs` file, given this is part of the public api

Reply:
My idea was to put it into the rpc protocol and re-export in the blobs.rs, but this also works. I guess it is better when people click on source when browsing docs.rs.